portal-impl/src/com/liferay/portlet/usersadmin/util/UsersAdminImpl.java

```diff
@@ -729,6 +729,37 @@ else if (organizations != null) {
 	}
 
 	@Override
+	public List<Object> getOrganizationsAndUsers(Hits hits) {
+		List<Document> documents = hits.toList();
+
+		List<Object> results = new ArrayList<>(documents.size());
+
+		for (Document document : documents) {
```

Comment:
This code will not scale. Keep in mind that there are Liferay installations with millions of users and a query could potentially return them all. So this could would make millions of queries to the database and collect all the results in memory.

Instead of doing this, return the Hits and process them with an specialized view in the UI. In that view you can obtain the actual Organization or User objects from the database if needed, since it will only be done for the items being shown (usually just 20, since that's the default page length). See document-library-web's search_resources.jsp for an example.